client/src/components/ChannelFilter.tsx

```diff
@@ -0,0 +1,81 @@
+import Select, {SingleValue} from 'react-select';
+import {ChannelFilterProps} from '../lib/types';
+import React from 'react';
+import styled from 'styled-components';
+import {useCallback, useEffect, useState} from 'react';
+import {fetchData} from '../services/fetch';
+
+interface Channel {
+    id: string;
+    name: string;
+}
+
+interface Option {
+    label: string;
+    value: string;
+}
+
+const FilterDiv = styled.div`
+      min-width: 350px;
+      max-width: 50%;
+      flex-grow: 1;
+    `;
+
+const ChannelFilter = ({onSelected, selectedChannel}: ChannelFilterProps) => {
+    const allValue = {
+        label: 'Code Reviews For All Slack Channels',
+        value: 'all',
+    };
+    const [channelOptions, setChannelOptions] = useState([] as Option[]);
+    const [selectPlaceHolder, setSelectPlaceHolder] = useState(allValue.label);
+    const handleChannelSelectionChange = useCallback((data: SingleValue<Option>) => {
```

Comment:
Not a blocker, I am still learning about `useCallback`, but why are you using it here? From what I can tell, the `onSelected` state setter function will always change on each render, so this will never use the cached returned value.

Reply:
The use of `useCallback` is to create a reference to the callback function so that it can be use for dependencies.  if a function is declare without a call back and use it as a dependencies, it will create a new reference the code is render, this will cause side affect and the component can and will refresh because it think the callback reference have change.

Think of it as useState... the state only get change when new value cause the update  which trigger dependencies component to re-render.  The useCallback have similar effect.  In this case, the callback is depend on onSelected reference.  If onSelected reference stayed the same, the useCallback return the same reference so it will not trigger other chained reaction that depend on the callback.

Reply:
Yes, if onSelect is changed, it will create a new reference because of dependencies.  The onSelect is passed by the caller.  So if that got change due to state change at the parent, then entire component will get re-render.  If we forgot to add the onSelect as dependencies, if that onSelect handler get change in the parent, the callback function via useCallback would cause invalid call because it would use existing references.

On the App component that callback is coming from useState hook.  If that function cause dependencies render, that would cause unstable component state.  From that point of view, I'm not sure.   I just have to assume useState would provide a stable reference to setState function.